### PR TITLE
DAOS-8144 tests: Fix IndexError for tests w/o agent running.

### PR DESCRIPTION
**Pull request by phender**
Do not attempt to log a message to the server log when a test is not
running a daos_agent.

Test-tag: pr daily_regression

Signed-off-by: Phillip Henderson <phillip.henderson@intel.com>

### CODE REVIEW

**Comment by mallove79** [comment]
> Do not attempt to log a message to the server log when a test is not
> running a daos_agent.
> 
> Test-tag: pr daily_regression

So the `pr` and `daily_regression` tags specified here will [run some agent-less tests](https://build.hpdd.intel.com/job/daos-stack/job/daos/view/change-requests/job/PR-6253/2/)?

**Review by mallove79 — DISMISSED** [review]

**Comment by phender** [comment]
In https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-6253/2/ the `cart_ctl` `--cfg_path` argument used the avocado `output_dir`  path which failed for these tests:

- Test / Functional on CentOS 7 / FTEST_checksum.CsumContainerValidation.1-./checksum/csum_basic.py:CsumContainerValidation.test_single_object_with_checksum;single-properties-single-hosts-mode_RW-createset-createsize-createsvc-1Byte-server_config-99f0
- Test / Functional on CentOS 7 / FTEST_checksum.CsumContainerValidation.2-./checksum/csum_basic.py:CsumContainerValidation.test_single_object_with_checksum;single-properties-single-hosts-mode_RW-createset-createsize-createsvc-1KB-server_config-d2f6
- Test / Functional on CentOS 7 / FTEST_checksum.CsumContainerValidation.3-./checksum/csum_basic.py:CsumContainerValidation.test_single_object_with_checksum;single-properties-single-hosts-mode_RW-createset-createsize-createsvc-Mixed-server_config-8bda
- Test / Functional on CentOS 7 / FTEST_control.DaosControlConfigTest.11-./control/daos_control_config.py:DaosControlConfigTest.test_daos_control_config_basic;transport_config-ca_cert_file_does_not_exist-transport_config-hosts-transport_config-e689
- Test / Functional on CentOS 7 / FTEST_control.DaosControlConfigTest.13-./control/daos_control_config.py:DaosControlConfigTest.test_daos_control_config_basic;transport_config-cert_file_does_not_exist-transport_config-hosts-transport_config-dfd0
- Test / Functional on CentOS 7 / FTEST_control.DaosControlConfigTest.15-./control/daos_control_config.py:DaosControlConfigTest.test_daos_control_config_basic;transport_config-key_file_does_not_exist-transport_config-hosts-transport_config-4dba
- Test / Functional on CentOS 7 / FTEST_datamover.DmvrPosixTypesTest.1-./datamover/dm_posix_types.py:DmvrPosixTypesTest.test_dm_posix_types_fs_copy;container-client_processes-dfuse-client_processes-hosts-client_processes-pool-server_config-dd19
- Test / Functional on CentOS 7 / FTEST_pool.PoolSvc.1-./pool/pool_svc.py:PoolSvc.test_pool_svc;hosts-pool-server_config-svc_params_none-317c

Due to the path being an invalid parameter, e.g.
```
2021-07-21 23:55:50,279 process          L0626 INFO | Running 'cart_ctl add_log_msg --cfg_path /var/tmp/ftest/avocado/job-results/job-2021-07-21T23.54-75ecd4f/test-results/1-._checksum_csum_basic.py_CsumContainerValidation.test_single_object_with_checksum_single-properties-single-hosts-mode_RW-createset-createsize-createsvc-1Byte-server_config-99f0/data --group-name daos_server -m "Test.name: 1-./checksum/csum_basic.py:CsumContainerValidation.test_single_object_with_checksum;single-properties-single-hosts-mode_RW-createset-createsize-createsvc-1Byte-server_config-99f0" --rank all'
2021-07-21 23:55:50,386 process          L0458 DEBUG| [stderr] 07/21-23:55:50.38 wolf-105vm1 CaRT[27489/27489/0] CLOG EMRG src/utils/ctl/cart_ctl.c:579 ctl_init() Assertion 'rc == 0' failed: crt_group_config_path_set() failed, DER_INVAL(-1003): 'Invalid parameters'
```

The avocado `output_dir`  path did work for other tests however, e.g.
```
2021-07-22 00:30:21,564 process          L0626 INFO | Running 'cart_ctl add_log_msg --cfg_path /var/tmp/ftest/avocado/job-results/job-2021-07-22T00.29-d919980/test-results/01-._container_basic_snapshot.py_BasicSnapshot.test_basic_snapshot_hosts-OC_S1-size1gb-server_config-4030/data --group-name daos_server -m "Test.name: 01-./container/basic_snapshot.py:BasicSnapshot.test_basic_snapshot;hosts-OC_S1-size1gb-server_config-4030" --rank all'
2021-07-22 00:30:25,942 process          L0714 INFO | Command 'cart_ctl add_log_msg --cfg_path /var/tmp/ftest/avocado/job-results/job-2021-07-22T00.29-d919980/test-results/01-._container_basic_snapshot.py_BasicSnapshot.test_basic_snapshot_hosts-OC_S1-size1gb-server_config-4030/data --group-name daos_server -m "Test.name: 01-./container/basic_snapshot.py:BasicSnapshot.test_basic_snapshot;hosts-OC_S1-size1gb-server_config-4030" --rank all' finished with 0 after 4.374885082244873s
```

I wonder if the length of the path is the issue.  For now I've reverted back to copying the file to the current working directory and using `--cfg_path .`

File: src/tests/ftest/util/apricot/apricot/test.py [hunk 1]
```diff
@@ -701,36 +701,40 @@ def setUp(self):
...
-        cart_ctl.rank.value = "all"
-        cart_ctl.cfg_path.value = "."
-        cart_ctl.group_name.value = "daos_server"
-        cart_ctl.m.value = s
+    def write_string_to_logfile(self, message):
+        """Write a string to the server log.
 
-        cart_ctl.n.value = None
-        cart_ctl.run()
+        The server log message will be appear in the following format:
```
Comment (shimizukko) [inline src/tests/ftest/util/apricot/apricot/test.py:710]:
```suggestion
        The server log message will appear in the following format:
```

**Review by shimizukko — APPROVED** [review]

**Comment by phender** [comment]
In https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-6253/3/ there was on failure:

- Test Hardware / Functional Hardware Medium / FTEST_osa.OSAOfflineReintegration.4-./osa/osa_offline_reintegration.py:OSAOfflineReintegration.test_osa_offline_reintegration_with_rf;transport_config-aggregation-blank_node-checksum-container-transport_config-hosts-clientslots-iorflags-loop_test-client_processes-32K-pool-rebuild-0-1-transport_config-setup-snapshot-test_obj_class-test_rf-b42c - https://jira.hpdd.intel.com/browse/DAOS-7948

File: src/tests/ftest/util/apricot/apricot/test.py (hunk 1 above)
Comment (phender) [reply src/tests/ftest/util/apricot/apricot/test.py:710]: Thanks, I'll address this change in a follow-on PR.

**Comment by phender** [comment]
> > Do not attempt to log a message to the server log when a test is not
> > running a daos_agent.
> > Test-tag: pr daily_regression
> 
> Will the `pr` and `daily_regression` tags specified here will [run some agent-less tests](https://build.hpdd.intel.com/job/daos-stack/job/daos/view/change-requests/job/PR-6253/2/)?

Yes, these tests that have failed in the past two daily runs passed in this PR:

- https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-6253/3/artifact/Functional%20on%20CentOS%207/control/dmg_nvme_scan_test.py/
- https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-6253/3/artifact/Functional%20on%20CentOS%207/server/daos_server_dump.py/

**Review by jolivier23 — APPROVED** [review]

**Pull request merged**